homeassistant/components/lyric/climate.py

```diff
@@ -223,16 +223,26 @@ def target_temperature(self) -> float | None:
     def target_temperature_high(self) -> float | None:
         """Return the highbound target temperature we try to reach."""
         device = self.device
-        if device.changeableValues.autoChangeoverActive:
-            return device.changeableValues.coolSetpoint
+        if (
+            device.changeableValues.autoChangeoverActive
+            and HVAC_MODES[device.changeableValues.mode] != HVAC_MODE_OFF
+        ):
+            if device.changeableValues.autoChangeoverActive:
+                return device.changeableValues.coolSetpoint
+            return None
```

Comment:
It would be cleaner to write this code like this:
```suggestion
        if (
            device.changeableValues.autoChangeoverActive
            and HVAC_MODES[device.changeableValues.mode] == HVAC_MODE_OFF
        ):
            return None
        if device.changeableValues.autoChangeoverActive:
            return device.changeableValues.coolSetpoint
            
```

Reply:
I agree, thanks for the suggestion, will make the changes now.

Reply:
In general it is better to reduce indented code if possible by using guard clause: https://deviq.com/design-patterns/guard-clause

I agree that there is a change in behavior in my suggestion, but I had in mind that you want to return `None` only in the case that `autoChangeoverActive` is active and HVAC mode is OFF.

if that is not the case you can also change the if to something like:
```python
        if (
            not device.changeableValues.autoChangeoverActive
            or HVAC_MODES[device.changeableValues.mode] == HVAC_MODE_OFF
        ):
            return None        
``` 
and in this case you don't need the additonal `if device.changeableValues.autoChangeoverActive` and you can return `device.changeableValues.coolSetpoint` after the `if`.

If my suggestion above does not match the behavior of the device, try to rewrite the code part by doing a first if and return. There are also two `if` one after another (which I missed on the first review) so at least these could be consolidated under one main `if` e.g:
```python
        if (
            device.changeableValues.autoChangeoverActive
            and HVAC_MODES[device.changeableValues.mode] != HVAC_MODE_OFF
        ):
            return device.changeableValues.coolSetpoint
```

Reply:
Thanks so much for the pointers. I have a background in coding but I'm new to python, so this really helps. I really like the first option, so I'll go with that. Thanks again!